src/components/PopupWithForm.js

```diff
@@ -1,11 +1,22 @@
 import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
-  constructor(popupSelector, handleFormSubmit) {
+  constructor(popupSelector, handleFormSubmit, shouldResetBeforeSubmit = true) {
     super(popupSelector);
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
     this._inputList = this._popupElement.querySelectorAll(".modal__input");
+    this._submitButton = this._popupForm.querySelector(".modal__button");
+    this._initialButtonText = this._submitButton.textContent;
+    this._shouldResetBeforeSubmit = shouldResetBeforeSubmit;
+  }
+
+  // Override open to reset form
+  open() {
+    if (this._shouldResetBeforeSubmit) {
+      this._popupForm.reset();
```

Comment:
You need to **reset** the form only   in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.